schema.graphql

```diff
@@ -813,10 +820,10 @@ type NameSignal @entity {
   unsignalledTokens: BigInt!
   "Tokens the curator has withdrawn from a deprecated name curve"
   withdrawnTokens: BigInt!
-  "Signal that the curator has from signaling their GRT"
+  "Shares of the name pool (GNS) that the curator has from signaling their GRT"
   nameSignal: BigInt!
-  # Note , we don't show vSignal here. We could, but it has almost no value to display it.
-  # We can get away with using curator nameSignal, total n and v signal on the subgraph.
+  "Actual signal shares that the name pool minted with the GRT provided by the curator"
+  signal: BigInt!
```

Comment:
this is another DYNAMIC value. if someone else starts minting alongside the GNS signal/vSignal, they could end up with negative signal here. or could sell all their nameSignal and still have signal here, even though thats not true. 

how do we plan to use this?

Reply:
This was mostly geared towards being able to track `totalSignal` and `totalAverageCostBasis` on the `Curator` entity, since what we had before were 2 different fields for `totalAverageCostBasis` and `totalSignal`, one for the curation done through name signals, and one for the curation done through version signals.

The idea was to get the actual amount of vSignal minted each time someone signalled through GNS, so that we can use the actual version signal to aggregate towards the `totalAverageCostBasis` and `totalSignal` field of the `Curator` for the version signal. (since 1 vSignal != 1nSignal).

This was intended to be able to have a "Total signal" and "total GRT on curation" (can't remember how it was named), in the curators table (so the totals were the totals for a given Curator/GraphAccount, across all the subgraphs)

Reply:
i believe it would be fine until there was a combination of :

- someone signals directly on the subgraph deployment, outside of the GNS
- the owner upgrades. then the signals will be changed

I think, then , how to fix it after this scenario, would be a batch update to take the new ratio of `nSignal to vSiganl` and apply it to every curator. But it might get a bit funky since `totalSignal` is an aggregation of many `signals` from many subgraphs

Reply:
After revisiting all cases I can think of, I agree that the only way this signal amount would be incorrect is in cases where a GNS signal migration occurs (because of an upgrade) and only if there's more GRT in the curve that was added directly through the curation contracts.

That would make the GNS unsignalling return more GRT than the original amount (since it basically would "rug pull" those that came after GNS), and thus, when signalling on the new version, the amount of version signal minted would be greater, but the amount of name signal would still be the same.

The fix would be the signals ratio change and batch update (as you mentioned), and I think it can be done safely since we can subtract the old version signal value that we had on the `NameSignal` entity, from the `Curator` total signal, and then reapply the new value we calculate based on the new nSignal/vSignal ratio.

Reply:
Yes I think it can be done as well. if you are up for it, lets try it!

Reply:
This should be fixed with the new changes, following the solution described above.